app/controllers/institutions_controller.rb

```diff
@@ -38,7 +45,28 @@ def ensure_page_size_is_positive
     render status: :unprocessable_entity, json: { message: 'invalid page size' } unless page_size.positive?
   end
 
+  def pg_name_querry
+    # a string
+    @pg_name_querry ||= params.fetch(:pg_name_querry, nil)
+  end
+
+  def pg_area_querry
+    # a string
+    @pg_area_querry ||= params.fetch(:pg_area_querry, nil)
+  end
+
   def page_size
+    # an integer
     @page_size ||= params.fetch(:page_size, 10).to_i
   end
+
+  def school_status
+    # boolean value
+    @school_status ||= params.fetch(:school_status, nil)
```

Comment:
This name can be confusing; something like `public_school` would probably be a better fit for a boolean.